heat/core/linalg/lanczos.py

```diff
@@ -0,0 +1,234 @@
+import math
+import torch
+
+from ..communication import MPI
+from .. import arithmetics
+from .. import dndarray
+from .. import factories
+from .. import random
+from .basics import matmul
+from .basics import dot
+
+__all__ = ["cg", "lanczos", "norm", "projection"]
+
+
+def cg(A, b, x0, out=None):
```

Comment:
Shouldn't this be in its own optimize module?

Reply:
It's not optimization that is happening here. We (@coquelin77 and me) were unsure where to put the functions with the new linalg module design, so for now we made a seperate file. I can have a look and refactor

Reply:
Yeah it is not optimization. But cg can do optimization or even better solving. Shouldnt cg go in a separate solver module?

Reply:
probably, it was, to be honest, just a byproduct/ warm up excersize for lanczos... but I guess it would be lonely in a seperate module.

Reply:
well, if we stick with scipy's places, linalg.cg makes sense

Reply:
ok, will do

Reply:
See above, is now linalg.solver